### ISSUE TITLE
Strange errors with refering to other components

### ISSUE THREAD

**Issue by avdeev** [styled-components/stylelint-processor-styled-components#54]
```
import styled from 'styled-components';
import { div } from 'react-dom';

const IconWrapper = styled.span`
  display: inline-block;
  ${div} & {
    color: #fff;
  }
`;

export default IconWrapper;
```

```
app/components/LeftMenu/IconWrapper.js
 6:22  ✖  Expected newline after ";" in a multi-line declaration block   declaration-block-semicolon-newline-after
 6:23  ✖  Expected empty line before rule                                rule-empty-line-before
```

**Comment by ismay** [comment]
Seems like the interpolation might be messing up the source map. I think what it's expecting is: 

```
const IconWrapper = styled.span`
  display: inline-block;

  ${div} & {
    color: #fff;
  }
`;
```

Does that fix the error?

Also, why aren't you writing:

```
const IconWrapper = styled.span`
  display: inline-block;

  div & {
    color: #fff;
  }
`;
```

I've never seen anyone use the syntax that you're using here.

**Comment by avdeev** [comment]
@ismay 
+1 error

```
 10:3   ✖  Unexpected empty line before declaration                       declaration-empty-line-before
```

This is just an example. In place of $ {div} there could be a custom component

**Comment by mxstbr** [comment]
Yeah we haven't built anything to support that newly introduced syntax yet, dang. We'll need to replace referenced components differently somehow...

**Comment by lgraziani2712** [comment]
Another error:

```js
const Link = styled.a`
  text-decoration: none;
  ${LinkContainer}:hover > a {
    font-weight: bold;
  }
`;
```

It throws: 

> ×  Expected -styled-mixin0 to come before text-decoration         order/properties-alphabetical-order
> ×  Expected newline after ";" in a multi-line declaration block      declaration-block-semicolon-newline-after

**Comment by emilgoldsmith** [comment]
Hey @lgraziani2712 yeah sorry about that. The different error is simply because of a changed implementation in a recent fix #59 which handled some other issues.

We are aware of the problem and if no one else beats me to it I'll probably implement something like what I proposed in #68 which would allow you to handle this :). Feel free to comment on that issue if you have any thoughts or feedback on the best way to structure that API!

**Comment by niftylettuce** [comment]
For anyone that encounters this issue, you can use this in the meanwhile:

```js
/* stylelint-disable-next-line declaration-empty-line-before, declaration-block-semicolon-newline-after */
${Button} {
  display: flex;
}
```

**Comment by emilgoldsmith** [comment]
See #34 last comment

**Comment by mirovelk** [comment]
Was this and #34 actually resolved? Because contextual selectors still throw errors for me and I am using version 1.2.2.

**Comment by emilgoldsmith** [comment]
Did you follow the last comment and use an interpolation tag?

**Comment by mirovelk** [comment]
Oh now I get it. I didn't dig that deep into the discussion. The tag works. Kind of a workaround but I am sure there is a good reason why it currently works like that. 

For everyone else and future reference, use `sc-selector` tag like this to prevent the error:

```
${/*sc-selector*/Wrapper}:hover & {
    whatever: here;
 }
```

Thank you for super fast response.

**Comment by chinesedfan** [comment]
> However, this solution only works when there's only one child Styled Component.

@daviddelusenet The reason is that `sc-selector` is always replaced with `div`. I tries to fix it as dynamic selectors in #219, i.e. `.sc-selector1`, `.sc-selector2`.
